### PR TITLE
Fix script class icons looking for paths at runtime

### LINKED ISSUE #25171: Exporting a game with a custom class with icon causes a crash
screenshot 1

**Godot version:**
3.1 beta 1 and beta 2

**OS/device including version:**
Windows 10 .exe file

**Issue description:**
When making a custom class with class_name syntax, the game wont run when exported with a custom icon. Works fine when not using a custom icon. Project runs fine from within editor also.
When running the exported .exe file in Windows 10, the game crashes. Not tested for other export targets.

**Steps to reproduce:**
Run the attached project, export project and try to run it. 

**Minimal reproduction project:**
[IconBug.zip](https://github.com/godotengine/godot/files/2776932/IconBug.zip)

### PR DESCRIPTION
**Pull request by willnationsdev**
I believe this closes #25171. At the very least, I made this change, re-compiled, re-generated export templates, exported the project, and encountered no errors.

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -3482,11 +3482,13 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 						Variant constant = tokenizer->get_token_constant();
 						String icon_path = constant.operator String();
 
+#ifdef TOOLS_ENABLED
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp]: Does any of the surrounding logic make sense in an exported game? IMO the GDScript parser should just discard all this editor-specific information in the runtime.

On top of checking `TOOLS_ENABLED`, this code should likely also check `Engine::get_singleton()->is_editor_hint()`, as the code is also not relevant when running the game with a `TOOLS_ENABLED` binary but not in the editor (or is it, for things like the remote debugger?).

Comment (vnen) [reply modules/gdscript/gdscript_parser.cpp]: The parser still needs to consume the tokens, but I agree that it doesn't need to fetch nor store the information.

Comment (willnationsdev) [reply modules/gdscript/gdscript_parser.cpp]: I have now updated it to wrap everything but the if statements and `advance` operations in `TOOLS_ENABLED` and an `Engine::get_singleton()->is_editor_hint()` check.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**